File: easy/1700_number_of_students_unable_to_eat_lunch/solution.cpp
#include <vector>

class Solution {
public:
  int countStudents(std::vector<int> &students, std::vector<int> &sandwiches) {
    std::vector<int> preferenceCounter(2, 0);

    for (const int &preference : students)
      ++preferenceCounter[preference];

    for (const int &sandwich : sandwiches) {
      if (preferenceCounter[sandwich]-- == 0)
        return preferenceCounter[!sandwich];
    }

    return 0;
  }
};

File: easy/1700_number_of_students_unable_to_eat_lunch/solution.py
from collections import Counter
from typing import List


class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        preference_counter = Counter(students)

        for sandwich in sandwiches:
            if preference_counter[sandwich] == 0:
                return preference_counter[not sandwich]

            preference_counter[sandwich] -= 1

        return 0
